Documentation/devel/distribution.md

```diff
@@ -0,0 +1,145 @@
+`cimd:aci-archive:v=0:https%3A%2F%2Fexample.com%2Fapp.aci`
+
+**Docker**
+This is a Name based distribution
+
+Docker defines a distribution using a docker registry
+The format is the same as the docker image string format (man docker-pull) with the "docker" scheme:
+`cimd:docker:v=0:[REGISTRY_HOST[:REGISTRY_PORT]/]NAME[:TAG|@DIGEST]`
+Examples:
+`cimd:docker:v=0:busybox`
+`cimd:docker:v=0:busybox:latest`
+`cimd:docker:v=0:registry-1.docker.io/library/busybox@sha256:a59906e33509d14c036c8678d687bd4eec81ed7c4b8ce907b888c607f6a1e0e6`
+
+### Future distributions
+
+**OCI Image distributions**
+This is a Name based distribution
+
+Now a docker distribution can also provide oci image but in future oci image will define one or more own kind of distribution starting from an image name (and additional tags, labels)
+
+**OCI Image layout**
+This is a Location based distribution
+
+It can fetch an image starting from a [OCI image layout](https://github.com/opencontainers/image-spec/blob/master/image-layout.md) format. The location can point to a single file archive, to a local/remote directory based layout or other kind of locations
+
+Probably it will be the sub-distribution used by the OCI image distribution described above) (like ACIArchive is the sub-distribution for the appc distribution).
+
+`cimd:oci-image-layout:v=0:file%3A%2F%2Fabsolute%2Fpath%2Fto%2Ffile?ref=refname`
+`cimd:oci-image-layout:v=0:https%3A%2F%2Fdir%2F?ref=refname`
+
+Since the OCI image layout can provide multiple images selectable by a ref there's the need to specify which ref to use in the archive distribution URI (see the above ref query parameter). Since distribution just covers one image it's not possible to import all the refs with just a distribution URI.
+
+TODO(sgotti). Define if oci-image-layout should internally handle both archive and directory based layouts or use two different distributions or a query parameter the explicitly define the layout (to avoid guessing if the url points to a single file or to a directory).
+
+**Note** Considering [this OCI image spec README section](https://github.com/opencontainers/image-spec#running-an-oci-image), probably the final distribution format will be similar to the app distribution. So there's a need to distinguish their User Friendly string (prepending an appc: or oci: ?).
+
+To sum it up:
+
+| Distribution   | Type     | URI Format                                                                | Sub-distribution |
+|----------------|----------|---------------------------------------------------------------------------|------------------|
+| Appc           | Name     | `cimd:appc:v=0:name?label01=....&label02=...`                             | ACIArchive       |
+| Docker         | Name     | `cimd:docker:v=0:[REGISTRY_HOST[:REGISTRY_PORT]/]NAME[:TAG&#124;@DIGEST]` | <none>           |
+| ACIArchive     | Location | `cimd:aci-archive:v=0:ArchiveURL?query...`                                |                  |
```

Comment:
I can't conceptually see the big difference between a final location or a name. As a user, there are just strings to locate my images (maybe not everything thinks this way).

@sgotti I am not sure about the Type: Name vs Type: Location thing. At this point I can't see why Distribution is not enough to express the Name vs Location ?

My other point is since we already express the difference in the `Distribution` field, unless we combine Distribution: Appc with Distribution: ACIArchive, and Distribution: OCI with Distribution: OCIImageLayout. I don't see why a separate `Location` field is needed?